muppets/src/websnapper.ts

```diff
@@ -81,23 +81,41 @@ function s3UploadFile(data: string | void | Buffer, filename: string, prefix: st
         console.log("Key = [%s]", putObjectPromise.Key);
         return putObjectPromise.Key;
     }).catch(err => {
+        if (params.Key.endsWith("screenshot.png")||params.Key.endsWith("screenshot.webp")){
+            throw new Error(`Upload failed for screenshot file [${params.Key}] : [${JSON.stringify(err)}]`);
+        } else if (params.Key.endsWith("html")) {
+            throw new Error(`Upload failed for html file [${params.Key}] : [${JSON.stringify(err)}]`);
+        } else if (params.Key.endsWith("har")) {
+            throw new Error(`Upload failed for har file [${params.Key}] : [${JSON.stringify(err)}]`);
+        }
         throw new Error(`Upload failed for file [${params.Key}] : [${JSON.stringify(err)}]`);
     });
 }
+export interface S3UploadFunction {
```

Comment:
- Don't name a function with a `Function` suffix
- Try to either:
  - export the function signature (`export type S3UploadFunction(data: string | void | Buffer /* etc */): Promise<string>`)
  - create a interface with a member function (e.g. `export interface IFileUploader`)